### PR TITLE
load DLLs only from safe load list

### PR DESCRIPTION
**Pull request by bratpiorka**
Set /DEPENDENTLOADFLAG to 2000 (LOAD_LIBRARY_SAFE_CURRENT_DIRS) in Windows umf.dll builds (MSVC only). 
With this flag, loading a DLL for execution from the current directory is only allowed if it is under a directory in the Safe load list.

In addition, there is a new test written in Power Shell that uses dumpbin.exe to check that this flag is set correctly.

Detailed information about the flag can be found at: https://learn.microsoft.com/en-us/cpp/build/reference/dependentloadflag?view=msvc-170
and: https://learn.microsoft.com/en-us/windows/win32/api/libloaderapi/nf-libloaderapi-loadlibraryexa

- [x] requires #534

### CODE REVIEW

File: .github/scripts/check_dll_flags.ps1 [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+$vcvarsall = "${vsPath}\vcvarsall.bat"
+echo "Visual Studio path: $vsPath"
+echo "vcvarsall.bat path: $vcvarsall"
+
+# Call vcvarsall.bat so we can run MSVC commands
+echo "Setting up MSVC environment..."
+Invoke-CmdScript "$vcvarsall" x86
+
+echo "$args"
+echo "$($args[0] | Out-String)"
```
Comment (PatKamin) [inline .github/scripts/check_dll_flags.ps1:34]: Don't we have to set and check this flag also for the umf_proxy.dll?

Comment (bratpiorka) [reply .github/scripts/check_dll_flags.ps1]: done

File: cmake/helpers.cmake [hunk 2]
```diff
@@ -136,11 +136,17 @@ function(add_umf_target_link_options name)
             endif()
         endif()
     elseif(MSVC)
+        message(STATUS "C_COMPILER_ID: ${C_COMPILER_ID}")
+        message(STATUS "CXX_COMPILER_ID: ${CXX_COMPILER_ID}")
+        message(STATUS "CMAKE_C_COMPILER_ID: ${CMAKE_C_COMPILER_ID}")
+        message(STATUS "CMAKE_CXX_COMPILER_ID: ${CMAKE_CXX_COMPILER_ID}")
```
Comment (PatKamin) [inline cmake/helpers.cmake:142]: nit: These messages potentially can be useful for other CMake generators. Could you move them to some common space, maybe in the main CMakeLists.txt?

**Review by PatKamin — APPROVED** [review]

File: .github/scripts/check_dll_flags.ps1 [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+echo "vcvarsall.bat path: $vcvarsall"
+
+# Call vcvarsall.bat so we can run MSVC commands
+echo "Setting up MSVC environment..."
+Invoke-CmdScript "$vcvarsall" x86
+
+# Get umf.dll configuration flags and check if DEPENDENTLOADFLAG is set to 0x2000
+$flags = & "${env:VCToolsInstallDir}\bin\Hostx64\x64\dumpbin.exe" /LOADCONFIG "$args"
+if (($flags | Where-Object { $_ -match '(\d+).*Dependent' } | ForEach-Object { $matches[1] } ) -eq "2000") { 
+  echo "The DEPENDENTLOADFLAG is correclty set to 0x2000."
```
Comment (lplewa) [inline .github/scripts/check_dll_flags.ps1:36]: correctly

Comment (bratpiorka) [reply .github/scripts/check_dll_flags.ps1]: done

File: cmake/helpers.cmake (hunk 2 above)
Comment (bratpiorka) [reply cmake/helpers.cmake]: these were only debug prints - removed

**Review by lplewa — APPROVED** [review]

**Pull request merged**